src/iologindata.cpp

```diff
@@ -1061,18 +1083,34 @@ bool IOLoginData::hasBiddedOnHouse(uint32_t guid)
 	return db.storeQuery(fmt::format("SELECT `id` FROM `houses` WHERE `highest_bidder` = {:d} LIMIT 1", guid)).get();
 }
 
+const std::vector<VIPGroup>& IOLoginData::getVIPGroups(uint32_t accountId)
+{
+	std::vector<VIPGroup> groups;
+
+	DBResult_ptr result = Database::getInstance().storeQuery(
+	    fmt::format("SELECT `id`, `name`, `editable` FROM `account_vipgroups` WHERE `account_id` = {:d}", accountId));
+	if (result) {
+		do {
+			groups.emplace_back(result->getNumber<uint32_t>("id"), result->getString("name"),
+			                    result->getNumber<uint16_t>("editable") != 0);
+		} while (result->next());
+	}
+	return groups;
+}
+
 std::forward_list<VIPEntry> IOLoginData::getVIPEntries(uint32_t accountId)
 {
 	std::forward_list<VIPEntry> entries;
 
 	DBResult_ptr result = Database::getInstance().storeQuery(fmt::format(
-	    "SELECT `player_id`, (SELECT `name` FROM `players` WHERE `id` = `player_id`) AS `name`, `description`, `icon`, `notify` FROM `account_viplist` WHERE `account_id` = {:d}",
+	    "SELECT `id`, `player_id`, (SELECT `name` FROM `players` WHERE `id` = `player_id`) AS `name`, `description`, `icon`, `notify`, (SELECT GROUP_CONCAT(DISTINCT `group_id` SEPARATOR ',') FROM `account_vipgroup_entry` WHERE `entry_id` = `id`) AS `groupIds` FROM `account_viplist` WHERE `account_id` = {:d}",
```

Comment:
I would prefer if you made 2 queries instead, one to get all entries, and another to get all (entry, group) pair, then sort them instead of handling stringified numbers :P